### ISSUE TITLE
Unified interface for archimedean copulas

### ISSUE THREAD

**Issue by FriesischScott** [AnderGray/BivariateCopulas.jl#13]
Because of their generator based nature Archimedean Copulas can be represented by a simple unified interface. An example for the clayton copula can be found in this gist: https://gist.github.com/FriesischScott/35eb27a6899e3556debd77242edf5ec5

My proposal looks something like this:

```julia
ArchimedeanCopula <: AbstractCopula

struct ClaytonCopula <: ArchimedeanCopula
	φ::Real
end
```

The (inverse) generators and derivatives for the bivariate case are available from literature. While φ is usually used to denote the generator it might be confusing with the super/subscripts. Alternatively we could just use `generator`, `inverse_generator` and `generator_derivative`. I do prefer shorter notation though. Not sure.
```julia
function φ(x::Real, c::ClaytonCopula)
    return (1+x)^(−1/c.ϑ)
end

function φ⁻(x::Real, c::ClaytonCopula)
    return x^(-c.ϑ) - 1
end

function φ¹(x::Real, c::ClaytonCopula)
    α = 1/.ϑ
    return gamma(1 + α)/gamma(α) * (1 + x)^(-(1 + α))
end
```

From here we can define the (inverse) rosenblatt transformation as
```julia
# Generalized for any archimedean copula
function rosenblatt(M::AbstractMatrix, c::ArchimedeanCopula)
    u1 = M[1, :]
    u2 = M[2, :]

    return [u1'; (φ¹.(φ⁻.(u1, c) + φ⁻.(u2, c), c) ./ φ¹.(φ⁻.(u1, c), c))']
end

function inverse_rosenblatt(U::AbstractMatrix, c::ClaytonCopula)
    u1 = U[1, :]
    u2 = U[2, :]

    return [u1'; ((u1.^(-ϑ) .* (u2.^(-ϑ/(ϑ + 1)) .- 1) .+ 1).^(-1/ϑ))']
end
```
I assume there is a generalized version for the inverse using generators as well but I haven't found a good representation yet. The rosenblatt transform maps the copula to independent uniform random variables and can also be used for goodness of fit tests.

Sampling can then be done using the inverse rosenblatt transform
```julia
function sample(c::ArchimedeanCopula, n::Int64)
	U = rand(2, n)
	return inverse_rosenblatt(U, c)
end
```

Other functions to consider are the analytical formulas for kendalls tau, spearmans rho, etc.

@AnderGray  I could make a pull request with an example implementation for the clayton copula including tests etc.

**Comment by FriesischScott** [comment]
From here defining methods for densities etc should be straightforward.

**Comment by AnderGray** [comment]
@FriesischScott all this sounds good, and would be a very nice contribution to the package. I think with the pull request we should also use requires.jl to remove the dependency with PyPlot, since that was giving some issues.

**Comment by AnderGray** [comment]
Closed with PR #14
